src/plugins/data/common/index_patterns/types.ts

```diff
@@ -11,21 +11,77 @@ import { ToastInputFields, ErrorToastOptions } from 'src/core/public/notificatio
 // eslint-disable-next-line
 import type { SavedObject } from 'src/core/server';
 import { IFieldType } from './fields';
-import { RUNTIME_FIELD_TYPES } from './constants';
 import { SerializedFieldFormat } from '../../../expressions/common';
 import { KBN_FIELD_TYPES, DataViewField } from '..';
 import { FieldFormat } from '../../../field_formats/common';
 
 export type FieldFormatMap = Record<string, SerializedFieldFormat>;
 
-export type RuntimeType = typeof RUNTIME_FIELD_TYPES[number];
-export interface RuntimeField {
+export type RuntimeType = estypes.MappingRuntimeFieldType | 'composite';
+
+/**
+ * The RuntimeField that will be sent in the ES Query "runtime_mappings" object
+ * We extends the object until @elastic/elasticsearch supports "composite" type
+ * and its "fields" object.
+ *
+ * To simplify the consuming code we enforce the script to be `InlineScript` type (and not also `string`)
+ */
+export interface ESRuntimeField extends Omit<estypes.MappingRuntimeField, 'type' | 'script'> {
```

Comment:
I want to second @mattkime's concern about relying on ES Types here. This data gets serialized and persisted, so we need to be very careful making changes to it. If estypes package makes a breaking change to any of these types, we need to be aware of it and write a migration to handle it so existing Index Patterns load correctly.  Without using separating types, we'd be de-serializing the Index Pattern into the wrong type without having any idea and only tests that load older data would catch it.

Reply:
Ok makes sense I will update the type 👍

Reply:
Done in https://github.com/elastic/kibana/pull/110226/commits/759f102ef36d3dbdaa753631ec529ceec0c013c1